Lib/glyphsLib/builder/__init__.py

```diff
@@ -12,6 +12,177 @@
+    to_ufo_font_attributes(context, family_name)
+
+    # get the 'glyphOrder' custom parameter as stored in the lib.plist.
+    # We assume it's the same for all ufos.
+    first_ufo = next(iter(context.ufos.values()))
+    glyphOrder_key = PUBLIC_PREFIX + 'glyphOrder'
+    if glyphOrder_key in first_ufo.lib:
+        glyph_order = first_ufo.lib[glyphOrder_key]
+    else:
+        glyph_order = []
+    sorted_glyphset = set(glyph_order)
+
+    for glyph in font.glyphs:
+        add_glyph_to_groups(kerning_groups, glyph)
+        glyph_name = glyph.name
+        if glyph_name not in sorted_glyphset:
+            # glyphs not listed in the 'glyphOrder' custom parameter but still
+            # in the font are appended after the listed glyphs, in the order
+            # in which they appear in the source file
+            glyph_order.append(glyph_name)
+
+        for layer in glyph.layers.values():
+            layer_id = layer.layerId
+            layer_name = layer.name
+
+            assoc_id = layer.associatedMasterId
+            if assoc_id != layer.layerId:
+                if layer_name is not None:
+                    supplementary_layer_data.append(
+                        (assoc_id, glyph_name, layer_name, layer))
+                continue
+
+            ufo = context.ufos[layer_id]
+            ufo_glyph = ufo.newGlyph(glyph_name)
+            load_glyph(context, ufo_glyph, layer, glyph)
+
+    for layer_id, glyph_name, layer_name, layer_data \
+            in supplementary_layer_data:
+        ufo_font = context.ufos[layer_id]
+        if layer_name not in ufo_font.layers:
+            ufo_layer = ufo_font.newLayer(layer_name)
+        else:
+            ufo_layer = ufo_font.layers[layer_name]
+        ufo_glyph = ufo_layer.newGlyph(glyph_name)
+        load_glyph(context, ufo_glyph, layer_data, layer_data.parent)
+
+    for ufo in context.ufos.values():
+        ufo.lib[glyphOrder_key] = glyph_order
+        if propagate_anchors:
+            propagate_font_anchors(ufo)
+        to_ufo_features(context, ufo)
+        add_groups_to_ufo(ufo, kerning_groups)
+
+    for master_id, kerning in font.kerning.items():
+        load_kerning(context.ufos[master_id], kerning)
+
+    result = list(context.ufos.values())
+
+    instances = font.instances
+    if do_filter_instances_by_family:
+        instances = list(filter_instances_by_family(instances,
+                                                    instance_family_name))
+    instance_data = {'data': instances}
+
+    # the 'Variation Font Origin' is a font-wide custom parameter, thus it is
+    # shared by all the master ufos; here we just get it from the first one
+    varfont_origin_key = "Variation Font Origin"
+    varfont_origin = first_ufo.lib.get(GLYPHS_PREFIX + varfont_origin_key)
+    if varfont_origin:
+        instance_data[varfont_origin_key] = varfont_origin
+    if debug:
+        return clear_data(font)
```

Comment:
I think we need to remove this `debug` argument as well as this `clear_data` function.
They no longer works with the class-based interface, as clear_data expects a dict or a list from which values are popped and the ones that are left are thus "unused".